test/ngAnimate/animateSpec.js

```diff
@@ -292,6 +292,129 @@ describe("animations", function() {
       });
     });
 
+    describe('customFilter()', function() {
+      it('should clear the `customFilter` if no function is passed',
+        module(function($animateProvider) {
+          expect($animateProvider.customFilter()).toBeNull();
+
+          $animateProvider.customFilter(angular.noop);
+          expect($animateProvider.customFilter()).toEqual(jasmine.any(Function));
+
+          $animateProvider.customFilter({});
```

Comment:
Actually, I just realized that the current implementation allows to "clear" the filter/className by passing a non-Function/non-RegExp argument. So, if we want to throw, we need to still allow some special value (e.g. `null`/`undefined`) for unsetting the filter/className.